source/menu.cpp

```diff
@@ -21,17 +22,34 @@ void init_menu(gfxScreen_t screen)
     consoleSelect(currentConsole);
 }
 
-ConsoleMenu currentMenu;
 
 void menu_draw_string(const char* str, int pos_x, int pos_y, const char* color)
 {
     currentMenu.menuConsole.cursorX = pos_x;
     currentMenu.menuConsole.cursorY = pos_y;
-    printf("%s%s%s", color, str, CONSOLE_RESET);
-
+    
     gfxFlushBuffers();
 }
 
+void ui_menu_draw_string(const char* str, int pos_x, int pos_y, u32 color)
+{
+    setTextColor(color); // black
+    renderText(pos_x, pos_y, FONT_DEFAULT_SIZE, FONT_DEFAULT_SIZE, false, str);
+}
+
+void ui_menu_draw(const char *title, const char* footer, int back, int count, const char *options[]) {
+    setTextColor(0xFF00FF00);
+    renderText(0, 8, 0.7f, 0.7f, false, title);
+    for (int i = 0; i < count && i < (currentMenu.menuConsole.consoleHeight - 2); i++) {
+        ui_menu_draw_string(options[i], 1, 32+(i*12), 0xFF000000);
```

Comment:
lol; yes they are guesses as to offsets and height. The renderText function (as re-appropriated from 3ds-examples) does lots of clever glyph width and height calculations. I was thinking this could be re-aligned to your currentMenu x and y positions and multiplied internally, but then issues arise from the x value as the font is variable width.

I chose not to get too hung up on it, as going forward, as I would image this to be a stop-gap really. I agree swapping out for defines will help readability